spotbugs/src/main/java/edu/umd/cs/findbugs/ba/jsr305/ValidationSecurityManager.java

```diff
@@ -1,130 +0,0 @@
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
- */
-
-package edu.umd.cs.findbugs.ba.jsr305;
-
-import java.lang.annotation.Annotation;
-import java.security.Permission;
-
-import javax.annotation.CheckForNull;
-import javax.annotation.meta.TypeQualifierValidator;
-import javax.annotation.meta.When;
-
-final class ValidationSecurityManager extends SecurityManager {
-
-
-    static final ValidationSecurityManager INSTANCE = new ValidationSecurityManager();
-    final static ValidatorClassLoader VALIDATOR_LOADER = ValidatorClassLoader.INSTANCE;
-
-
-    static {
-        if (TypeQualifierValue.DEBUG_CLASSLOADING) {
-            new RuntimeException("Creating ValidationSecurityManager #").printStackTrace();
-        }
-
-    }
-
-    public static <A extends Annotation> When sandboxedValidation(A proxy, TypeQualifierValidator<A> v, @CheckForNull Object constantValue) {
-        if (performingValidation.get()) {
-            throw new IllegalStateException("recursive validation");
-        }
-
-        try {
-            performingValidation.set(Boolean.TRUE);
-            if (TypeQualifierValue.DEBUG_CLASSLOADING) {
-                System.out.println("Performing validation in thread " + Thread.currentThread().getName());
-            }
-            try {
-                When result = v.forConstantValue(proxy, constantValue);
-                if (!performingValidation.get()) {
-                    throw new IllegalStateException("performingValidation not set when validation completes");
-                }
-                return result;
-            } catch (ClassCastException e) {
-                Class<? extends Annotation> c = proxy.getClass();
-                System.out.println(c.getName() + " extends " + c.getSuperclass().getName());
-                for (Class<?> i : c.getInterfaces()) {
-                    System.out.println("  " + i.getName());
-                }
-                throw e;
-            }
-
-        } finally {
-            performingValidation.set(Boolean.FALSE);
-            if (TypeQualifierValue.DEBUG_CLASSLOADING) {
-                System.out.println("Validation finished in thread " + Thread.currentThread().getName());
-            }
-
-        }
-    }
-
-
-    @Override
-    public void checkPermission(Permission perm) {
-        if (TypeQualifierValue.DEBUG_CLASSLOADING) {
-            System.out.println("Checking for " + perm + " permission in thread " + Thread.currentThread().getName());
-        }
-        if (performingValidation.get() && inValidation()) {
-            SecurityException e = new SecurityException("No permissions granted while performing JSR-305 validation");
```

Comment:
Do I understand correctly that under some conditions, analyzing bytecode could result in some application classes being loaded in the SpotBugs process and potentially running arbitrary code?